pkg/reconciler/build/build.go

```diff
@@ -109,29 +125,151 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	}
 
 	// Get the Build resource with this namespace/name
-	if _, err := c.buildsLister.Builds(namespace).Get(name); errors.IsNotFound(err) {
+	build, err := c.buildsLister.Builds(namespace).Get(name)
+	if errors.IsNotFound(err) {
 		// The Build resource may no longer exist, in which case we stop processing.
 		logger.Errorf("build %q in work queue no longer exists", key)
 		return nil
 	} else if err != nil {
 		return err
 	}
 
-	// TODO(jasonhall): adopt the standard reconcile pattern.
-	// For Build this looks something like:
-	// podName := names.Pod(build)
-	// pod, err := c.podLister.Pods(build.Namespace).Get(podName)
-	// if IsNotFound(err) {
-	// 	desired := resources.MakePod(build)
-	// 	pod, err = c.kubeclientset.V1().Pods(build.Namespace).Create(desired)
-	// 	if err != nil {
-	// 		return err
-	// 	}
-	// } else if err != nil {
-	// 	return err
-	// }
-	//
-	// // Update build.Status based on pod.Status
+	// Don't mutate the informer's copy of our object.
+	build = build.DeepCopy()
+
+	// Validate build
+	if err = c.validateBuild(build); err != nil {
+		c.Logger.Errorf("Failed to validate build: %v", err)
+		return err
+	}
+
+	// If the build's done, then ignore it.
+	if !builder.IsDone(&build.Status) {
```

Comment:
feels worth inverting this:
```go
if builder.IsDone(&build.Status) {
  return nil
}
```

Reply:
Good call. I can also remove another layer of indentation!